### PR TITLE
Update helm chart to use the common-library

### PR DESCRIPTION
**Pull request by marcsanmi**
Resolves https://github.com/newrelic/nri-kube-events/issues/98.

I tried to keep every change separately in different (kind of) atomic commits.

### CODE REVIEW

**Review by gsanchezgavier — COMMENTED** [review]
looks good overall, Thanks! i left some comments

File: charts/nri-kube-events/templates/_helpers.tpl
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers.tpl:7]: You mean we can remove this https://github.com/newrelic/nri-kube-events/blob/98417b75b2501765344af7abe8b0884f9412e46f/charts/nri-kube-events/templates/deployment.yaml#L1 safely?

Comment (gsanchezgavier) [reply charts/nri-kube-events/templates/_helpers.tpl:7]: yes , clustername and license are checked by the library , the installation will fail if any of those are note definded

File: charts/nri-kube-events/Chart.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/Chart.yaml]: Updated in d85da9454b2cef86916c915a6a8c6262b5870929

File: charts/nri-kube-events/Chart.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/Chart.yaml]: Ditto

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Ditto

File: charts/nri-kube-events/tests/deployment_test.yaml
Comment (kang-makes) [inline charts/nri-kube-events/tests/deployment_test.yaml:33]:
```suggestion
           pattern: .*newrelic/nri-kube-events:latest$
```

**Review by kang-makes — CHANGES_REQUESTED** [review]
There are still things from the `common-library` that are still unimplemented.

If you take a [glance at the templates from the library](https://github.com/newrelic/helm-charts/tree/master/library/common-library/templates/) you'll see that there are things like `hostNetwork` and `dnsConfig` that are new which you still don't have in you PR. Keep in mind that there are also things that the library supports (like `lowDataMode`) that could not make sense in this chart.

The most needed is the secret for the license because it does a sanity check from the user's input.

To not make this PR bigger, after you address the comments that I left, I would leave [the refactor of the agent's configuration](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_agent-config.tpl) to a follow-up PR.

Besides me requesting changes this is a really good job. Keep it up!

File: charts/nri-kube-events/templates/deployment.yaml
Comment (kang-makes) [reply charts/nri-kube-events/templates/deployment.yaml]: Solved with [#cb86c8f](https://github.com/newrelic/nri-kube-events/pull/106/commits/cb86c8fab0daa008c166abf79b1c65454e561e5d)

File: go.mod
Comment (marcsanmi) [reply go.mod:188]: Going to update it in a separate PR.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Thanks for the context, this was updated here c6c698c709eccce1b7669c1e80493a1d2671e012

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Updated in 43f840e0a3f4367aca4975e51336a2108742e894.

File: charts/nri-kube-events/tests/deployment_test.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/tests/deployment_test.yaml]: Updated in 43f840e0a3f4367aca4975e51336a2108742e894

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: My bad, updated in 43f840e0a3f4367aca4975e51336a2108742e894

File: charts/nri-kube-events/templates/_helpers.tpl
Comment (gsanchezgavier) [inline charts/nri-kube-events/templates/_helpers.tpl:1]: same here, any reason to keep the file ?

File: charts/nri-kube-events/templates/NOTES.txt
Comment (gsanchezgavier) [inline charts/nri-kube-events/templates/NOTES.txt:1]: any reason to keep the file empty?

**Review by gsanchezgavier — APPROVED** [review]
looks good on my side , let's wait for a Juanma second review

File: charts/nri-kube-events/templates/NOTES.txt
Comment (marcsanmi) [reply charts/nri-kube-events/templates/NOTES.txt:1]: Nope, I thought it was "kind-of-our-standard" and even if empty, it was a good practice leave it. Going to updated.

File: charts/nri-kube-events/templates/_helpers.tpl
Comment (marcsanmi) [reply charts/nri-kube-events/templates/_helpers.tpl]: Updated in 8488d1d0d4b77194599e360989414f6fc7778b3d.

File: charts/nri-kube-events/templates/NOTES.txt
Comment (marcsanmi) [reply charts/nri-kube-events/templates/NOTES.txt]: Ditto.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (kang-makes) [inline charts/nri-kube-events/templates/deployment.yaml:130]:
```sugestion
         {{- . | nindent 8 }}
```

File: charts/nri-kube-events/templates/deployment.yaml
Comment (kang-makes) [inline charts/nri-kube-events/templates/deployment.yaml:126]:
```sugestion
         {{- . | nindent 8 }}
```
If you render `test: test` toleration with a right line cleaner it could render like this:
```yaml
tolerations:
  test:testaffinity:
```
It is not going to happen because of lines 133 and 134 but be really careful when to and not to put line chompers.

We usually, while rendering, put it only on the left.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (kang-makes) [inline charts/nri-kube-events/templates/deployment.yaml:122]: Same here.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (kang-makes) [reply charts/nri-kube-events/templates/deployment.yaml]: This is still templated a map. It is a string 😅 

```suggestion
      priorityClassName: {{ . }}
```

I you have doubts, take a look at [CHART-TEMPLATE](https://github.com/newrelic/helm-charts/blob/master/library/CHART-TEMPLATE/templates/integration-deployment.yaml#L29-L31).

File: charts/nri-kube-events/ci/test-values.yaml
Comment (kang-makes) [inline charts/nri-kube-events/ci/test-values.yaml:19]: NIT: Do this apply to this chart?

**Review by kang-makes — APPROVED** [review]
In PRs like this is a good idea to rebase everything to do the change in the commit where you did that change but it is not a good idea once you have submitted it because you lost the history of what you have already reviewed or not.

Just to point out why I took so much time on re-reviewing everything. Sorry for answering so late and the rant.

It is almost perfect, just left a few NITs here and there. LGTM.

Really good job :)

File: charts/nri-kube-events/templates/deployment.yaml
Comment (kang-makes) [reply charts/nri-kube-events/templates/deployment.yaml]: I said nothing. You said it was solved on other commit you force-pushed so I left the comment just in case it was not properly solved.

It is solved.

File: charts/nri-kube-events/ci/test-values.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/ci/test-values.yaml]: Nope, I don't know why I really add it... Updated in 1075616da92089379b613d64df96f24fbd82066b.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Updated in f956251cb7b1d52ab66d364bd839a2b861952ff7.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Ditto.

File: charts/nri-kube-events/templates/deployment.yaml
Comment (marcsanmi) [reply charts/nri-kube-events/templates/deployment.yaml]: Ditto.

**Review by kang-makes — APPROVED** [review]
> [marcsanmi](https://github.com/marcsanmi) dismissed [kang-makes](https://github.com/kang-makes)’s stale review via 1075616 2 hours ago

I thought we had that disabled... Nevermind. Approved.

**Pull request merged**
